api/src/middleware/mongoose.js

```diff
@@ -0,0 +1,28 @@
+const mongoose = require('mongoose');
+const { fillErrorObject } = require('../middleware/error');
+
+async function validateObjectId(_id, type, res, next) {
+  if (!mongoose.Types.ObjectId.isValid(_id)) {
+    console.log('in here');
+    console.log(_id);
+    next(
+      fillErrorObject(400, 'Validation error has occurred', [
+        'Given ' + type + ' id is not in a valid hexadecimal format',
+      ])
+    );
+  } else {
+    next();
```

Comment:
Does this not go to the error function?

Reply:
Yes, because there's no argument passed to next(), so it won't go to the error function, it'll go on to run the specified function in the controller file. If it passes the objectId validation, then there's no need to go to the error function.